packages/react-composites/src/composites/ChatComposite/adapter/AzureCommunicationChatAdapter.ts

```diff
@@ -224,10 +241,25 @@ export class AzureCommunicationChatAdapter implements ChatAdapter {
   off(event: 'participantsAdded', listener: ParticipantsAddedListener): void;
   off(event: 'participantsRemoved', listener: ParticipantsRemovedListener): void;
   off(event: 'topicChanged', listener: TopicChangedListener): void;
-  off(event: 'error', listener: (e: Error) => void): void;
+  off(event: 'error', listener: ChatErrorListener): void;
   off(event: ChatEvent, listener: (e: any) => void): void {
+    if (event === 'error') {
+      this.context.off(event, listener);
+      return;
+    }
     this.emitter.off(event, listener);
   }
+
+  private async asyncTeeErrorToEventEmitter<T>(f: () => Promise<T>): Promise<T> {
```

Comment:
The on('error') event is *not* generated from the error stored in state.

This is one example of a general idea: Any computation of events by diffing state will be lossy. We must allow state updates to be batched for performance reasons. Users of stateful clients only care about binding to the "current" state of the data. Thus, it is possible that an observer of state changes misses some intermediate transitions.